setuptools/package_index.py

```diff
@@ -588,24 +588,27 @@ def find(req, env=None):
                     continue
 
                 if dist in req and (dist.precedence<=SOURCE_DIST or not source):
-                    return dist
+                    mylocation = self.download(dist.location, tmpdir)
+                    if os.path.exists(mylocation):
+                        return dist, mylocation
+            return None, None
```

Comment:
No worries. I acknowledge - there are some old and undesirable patterns in setuptools, given its 11 year history.